backend/recipes/base.py

```diff
@@ -128,6 +128,8 @@ def __init__(self, plan):
         if 'title' in plan:
             self.currentRecipe = plan['title']
 
+        self._microlabHardware = MicroLabHardware.get_microlab_hardware_controller()
```

Comment:
(Copied from #267)

I think it would be better if recipe accepted microlabHardware as an parameter when creating the object, rather than loading the singleton for it here directly. That way for testing Recipe we can pass in whatever MicroLabHardware objects we want and don't need to deal with making sure the singleton is in the correct state.

Reply:
Sure, I can move it to Recipe class instantiation